### ISSUE TITLE
Change ut_dbms_output_suite_reporter to ut_documentation_reporter

### ISSUE THREAD

**Issue by jgebal** [utPLSQL/utPLSQL#101]
The reporter should provide text outputs in a format as below image:
![image](https://cloud.githubusercontent.com/assets/1398171/20240825/1fef1db4-a91a-11e6-8d21-42ada82d2396.png)

The items of high importance
1. Suites are reported by descriptive name (if avaliable)
Each sub-item (suite or test) is indented by with 2 spaces
2. There is a single empty line present between individual suites
3. If a test is failing, a text `(FAILED - x)` is shown next to the test name. `x` is a consecutive failure number
4. If suite is not named then a package name with owner name is used
If a test is not named, then a test procedure name is used (with brackets)
5. After all the tests were reported, and if any of tests failed, a `Failures:` section is present.
The section is at the bottom of the report, appearing only once, one for all suites executed in a run.
6. The `(FAILED - x)` tests are referenced in `Failures:` report in order of failure.
For each failed test a **full name** is built out of: root suite name, branch suites, test name.
For each failed test, a `Failures/errors:` section is shown.
The section contains information about all failed assertions.
The information contains information that will allow identifying why the assertion and test failed.
7.  For each failed assertion, a `owner.package:line_number` is used to help navigate to the failure
8. At the end of the report, information about run time 
9. Information about number of tests. IF any failed, number of failed is displayed too.
10. Brief summary of failed examples in a format:
owner.package_name:line_number # **full name**

**Comment by jgebal** [comment]
This is resolved by #109 
The formatting of  reporter output is documented by examples
```
Demo of documentation reporter
  repoting exception (FAILED - 1)
  UT3.DEMO_DOC_REPORTER1.failing_no_name (FAILED - 2)
  A failing test exsample (FAILED - 3)
  A passing test sample
  UT3.DEMO_DOC_REPORTER1.test_without_name

A suite pacakge without body
  A test (FAILED - 4)
  UT3.DEMO_DOC_REPORTER2.passing_test2 (FAILED - 5)

suite_package_without_name
  A passing test sample
  A passing test sample

Failures:
   1) repoting exception
      Failures/Errors:
	error: ORA-01858: a non-numeric character was found where a numeric was expected
	       ORA-06512: at "UT3.DEMO_DOC_REPORTER1", line 21
	       ORA-06512: at line 5

   2) UT3.DEMO_DOC_REPORTER1.failing_no_name
      Failures/Errors:
	expected: 2016-11-27T03:54:08(date)
	to equal: '27-NOV-16'(varchar2)
	error: The matcher 'equal' cannot be used for comparison of data type (varchar2) with data type (date).
	at "UT3.DEMO_DOC_REPORTER1", line 16 "ut.expect(sysdate).to_(equal(to_char(sysdate)));"

   3) A failing test exsample
      Failures/Errors:
	expected: 1(number)
	to equal: 2(number)
	at "UT3.DEMO_DOC_REPORTER1", line 12 "ut.expect(1).to_(equal(2));"

   4) A test
      Failures/Errors:
	error: ORA-04067: not executed, package body "UT3.DEMO_DOC_REPORTER2" does not exist
	       ORA-06508: PL/SQL: could not find program unit being called: "UT3.DEMO_DOC_REPORTER2"
	       ORA-06512: at line 5

   5) UT3.DEMO_DOC_REPORTER2.passing_test2
      Failures/Errors:
	error: ORA-04067: not executed, package body "UT3.DEMO_DOC_REPORTER2" does not exist
	       ORA-06508: PL/SQL: could not find program unit being called: "UT3.DEMO_DOC_REPORTER2"
	       ORA-06512: at line 5

Finished in .034766 seconds
9 tests, 5 failures
```

**Comment by Pazus** [comment]
PR was merged, close the issue.
